### ISSUE TITLE
hostname: invalid option -- 'I'

### ISSUE THREAD

**Issue by AdJaGu** [HorlogeSkynet/Archey4#4]
After updating to the most recent version I received the following error when running archey.

hostname: invalid option -- 'I'... etc...

I am on Arch Linux and on my machine` hostname` has an `'i'` option not an `'I'` option. `hostname -i` returns a value of 127.0.1.1 on my machine (personal home computer without web accessible servers).

The attached photograph shows the entire error message that I received.

![screenshot_2017-08-04_01-26-50](https://user-images.githubusercontent.com/6255623/28955475-14544f36-78b5-11e7-8b66-b185867f9633.png)

**Comment by HorlogeSkynet** [comment]
Hey @AdJaGu,

It looks like you're running an old version of `hostname`...
Run `hostname --version` and tell me what it says. On my side it outputs : `hostname 3.18`.

The version that added the `-I` option (`--all-ip-addresses`) should be greater than yours.

I start thinking about that, expect a workaround within 12 hours 😉

Greetings 👋

**Comment by AdJaGu** [comment]
Well, I am using an outdated version. `hostname --version` returns `1.9.4`.

**Comment by HorlogeSkynet** [comment]
Hey, thanks for the answer, so it looks like this package is even outdated in the AUR.
I'll try within some days to build a workaround for these older versions, but it would not be as fast as `hostname -I` is...

👋

**Comment by AdJaGu** [comment]
I do have `ip` available on the system. `ip -V` returns `ip utility, iproute2-ss170705`.

**Comment by HorlogeSkynet** [comment]
Perfect, tomorrow or monday I'll try to make something from it 😉, I'll keep you posted 👌

**Comment by AdJaGu** [comment]
Cool. I was surprised to find out that debian hostname actually has more options than GNU hostname (from GNU inetutils). One of the main reasons I moved to Arch Linux was the newer, updated packages. Thank you for your time and consideration @HorlogeSkynet.

**Comment by HorlogeSkynet** [comment]
Re' @AdJaGu !

As promised, you'll find the workaround I was talking about onto the new branch.
Could you clone and test it ?

When you'll be OK, I'll merge that and replace tabs by white-spaces as Python recommends it...

++

**Comment by AdJaGu** [comment]
All working on my end now @HorlogeSkynet.
![screenshot_2017-08-06_16-26-38](https://user-images.githubusercontent.com/6255623/29006762-28a8d3ce-7ac4-11e7-9405-9184cba8ceff.png)

**Comment by AdJaGu** [comment]
While I was right it ran without error, the displayed ip address is not the address of my machine. The ip address to this machine (router always gives this machine the same address because it hands addresses out based upon mac address) is `192.168.1.10`, but the script displays `192.168.1.255`.

Edited `ip addr` output follows:
```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: enp3s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether ??:??:??:??:??:?? brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.10/24 brd 192.168.1.255 scope global enp3s0
       valid_lft forever preferred_lft forever
    inet6 ????:????:????:????:????:????:????:???/64 scope global mngtmpaddr noprefixroute dynamic 
       valid_lft 604625sec preferred_lft 604625sec
    inet6 ????::????:????:????:????/64 scope link 
       valid_lft forever preferred_lft forever
3: wlp2s0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000
```

**Comment by normcyr** [comment]
The `192.168.1.255` address is probably the broadcast one. 

@AdJaGu would the output of `/sbin/ifconfig | grep "inet" | awk -F' ' '{print $2}' | sed -n 3p` give you your local IP?

**Comment by normcyr** [comment]
or if you do not have the `net-tools` package installed, would `ip route show` get your local IP in the output?

**Comment by HorlogeSkynet** [comment]
Hello to both of you @AdJaGu @normcyr,

Sorry, my bad : While re-adapting the `Popen` calls, I did choose a bad field number.
Now (with c04d74dd8ecd35cae4a6d5da30658a9274d4b712), it'll display the machine's IP address(es), not the network's broadcast one(s).

@normcyr `ip route show` will effectively display the IP address, but not the other ones like `hostname -I` did in your case ;)
Your trick with `ifconfig` is good, but the command is deprecated for now, that's why I'd rather use `ip` !

++

**Comment by AdJaGu** [comment]
@normcyr Arch Linux depreciated `ifconfig` in favor of `ip`, and yes, `ip route show` does have the correct source address shown.

@HorlogeSkynet I'll pull the new one and give it a go and return with results.

**Comment by HorlogeSkynet** [comment]
Hello @AdJaGu,

The `WAN_IP` now uses `dig +short myip.opendns.com @resolver1.opendns.com`, or (if you don't have `dig`) what this page displays : <https://ident.me>.

Could you tell me what it outputs in your case ?

It can't be related to the `LAN_IP` results, and if you have an IPv6 address, it's because your FAI gives you one 😮

Moreover, your machine is not connected to the Internet, right ? So it should displays `No Address` 🤔

**Comment by AdJaGu** [comment]
@HorlogeSkynet I do not have dig, but when I visited https://ident.me it did display this computers IPv6 address, same thing that Archey was showing (I say was because when I run it now it does not populate `WAN_IP` at all).

Yes I am connected to the internet, via my router. I believe I was confusing `WAN_IP` with WiFi earlier because this computer has both wired and wireless I for some reason was thinking that `WAN_IP` was for my wireless connection (which is never active).

So, now that I am no longer confused here is what happens. I run Archey and it works and returns the proper IPv4 (local) and IPv6 (external) addresses for this computer. I have zero idea why `WAN_IP` is not populating now lol.

**Comment by HorlogeSkynet** [comment]
I was doing the same mistake some years ago too 😆

So everything is OK I assume ? If it's the case, feel free to close this one, and I'll merge that !

**Comment by AdJaGu** [comment]
Yeah, all is good. Thank you for your time and effort to fix this @HorlogeSkynet  :+1:

**Comment by HorlogeSkynet** [comment]
Perfect 👌, merged with : c04d74dd8ecd35cae4a6d5da30658a9274d4b712 (fast-forward was unfortunately disabled...)

--> No problem, you're welcome. See you for the next issue @AdJaGu 😉

**Comment by smichaud** [comment]
> or if you do not have the `net-tools` package installed, would `ip route show` get your local IP in the output?

`ip route show | head --lines 1 | awk '{print $9}'` 

Worked perfectly on both Manjaro ethernet and Raspberry Pi OS WiFi.
